tests/test_url_build.py

```diff
@@ -32,12 +32,21 @@ def test_build_with_scheme_and_host():
     assert u == URL("http://127.0.0.1")
 
 
-def test_build_with_port():
-    with pytest.raises(ValueError):
-        URL.build(port=8000)
-
-    u = URL.build(scheme="http", host="127.0.0.1", port=8000)
-    assert str(u) == "http://127.0.0.1:8000"
+@pytest.mark.parametrize(
+    ("port", "exc", "match"),
+    [
+        pytest.param(
+            8000,
+            ValueError,
+            r'^Can\'t build URL with "port" but without "host".$',
```

Comment:
I imagine escaping `'` within a raw-string would produce `\\'`. Not sure what it'd do to the regexp…
In such cases, I tend to split the literal as follows:
```suggestion
            r"^Can't " r'build URL with "port" but without "host".$',
```

Reply:
Another (better?) option is to use a multiline raw-string combined with a `re.VERBOSE` flag:
```suggestion
            r"""(?x)
            ^
            Can't build URL with "port" but without "host"\.
            $
            """,
```

Reply:
As 0-3 \ infront of a ' are allowed in a re to match a '
```
>>> re.match("^test't$", "test't")
<re.Match object; span=(0, 6), match="test't">
>>> re.match("^test\'t$", "test't")
<re.Match object; span=(0, 6), match="test't">
>>> re.match("^test\\'t$", "test't")
<re.Match object; span=(0, 6), match="test't">
>>> re.match("^test\\\'t$", "test't")
<re.Match object; span=(0, 6), match="test't">
>>> re.match("^test\\\\'t$", "test't")
>>>
```
raw strings end up with 2 \, and any of the proposed changes to the re do not improve read-ability, I'd go with it?

Reply:
I disagree — the multiline raw-string is more readable, because in verbose mode, it allows moving the line start/end markers to separate lines. Moving complexity to other lines has a positive cognitive effect. I'd also argue that any escaped things in a string (not even a regexp) do reduce the readability.

Reply:
okay than